### PR TITLE
Updated pybind11

### PR DESCRIPTION
**Pull request by hakonhagland**
Updated pybind11 from version 2.3.0 to version 2.11.0.

Basically just replaced the folder `python/pybind11`: https://github.com/OPM/opm-common/tree/master/python/pybind11 with this tar ball:
https://github.com/pybind/pybind11/archive/refs/tags/v2.11.0.tar.gz

### CODE REVIEW

**Comment by hakonhagland** [comment]
jenkins build this please

**Comment by hakonhagland** [comment]
jenkins build this opm-simulators=4845 please

File: python/pybind11/README.rst [hunk 1]
```diff
@@ -0,0 +1,183 @@
+Note: This is a dump of the pybind11 distribution, version 2.11.0.
```
Comment (hakonhagland) [inline python/pybind11/README.rst:1]: This line was added to the original file in https://github.com/pybind/pybind11/archive/refs/tags/v2.11.0.tar.gz in accordance with the comment https://github.com/OPM/opm-common/pull/819#issuecomment-502545455

File: CMakeLists.txt [hunk 2]
```diff
@@ -267,7 +267,7 @@ if (OPM_ENABLE_PYTHON)
     endif()
     set(PYTHON_INCLUDE_DIRS ${PYTHON_INCLUDE_DIR})
   endif()
-  find_package(pybind11 2.2 CONFIG)
+  find_package(pybind11 2.11 CONFIG)
```
Comment (hakonhagland) [inline CMakeLists.txt:270]: Increased the version from 2.2 to 2.11 here

File: log.txt
Comment (bska) [inline log.txt:1]: What purpose does this file (`log.txt`) serve in this context?

Comment (hakonhagland) [reply log.txt:1]: Thanks, that was committed by accident. Removed now.

File: CMakeLists.txt (hunk 2 above)
Comment (akva2) [reply CMakeLists.txt:270]: that's a very bad idea. this is a minimum version, and a lot of us use system packages. i suggest keeping this at 2.2 and bump if we ever add something that actually needs the newer version, or at the very least only bump it to 2.4 which is the version in use on ubuntu 20.04.

Comment (hakonhagland) [reply CMakeLists.txt:270]: Agree,  so you install pybind11 as a system package then? I could not find the package on ubuntu, I tried `pip install pybind11` but cmake was not able to find that installation. Anyway I think the solution in https://github.com/OPM/opm-common/pull/3671 is better so I suggest we close this PR.

**Pull request closed without merging**